db/blob/blob_fetcher.h

```diff
@@ -32,6 +33,15 @@ class BlobFetcher {
 
  private:
   const Version* version_;
-  ReadOptions read_options_;
+  const ReadOptions& read_options_;
 };
+
+class BlobFetcherCopyReadOptions : public BlobFetcher {
```

Comment:
This naming `BlobFetcherCopyReadOptions` sounds like it's a type of `ReadOptions` than ` BlobFetcher`. How about `BlobFetcherWithReadOptionsCopy`.